concordium-node/src/grpc2.rs

```diff
@@ -985,6 +985,54 @@ pub mod server {
             let response = tonic::Response::new(receiver);
             Ok(response)
         }
+
+        #[cfg(feature = "network_dump")]
+        async fn dump_start(
+            &self,
+            request: tonic::Request<crate::grpc2::types::DumpRequest>,
+        ) -> Result<tonic::Response<crate::grpc2::types::BoolResponse>, tonic::Status> {
+            let file_path = request.get_ref().file.to_owned();
+            let result = self
+                .node
+                .activate_dump(
+                    if file_path.is_empty() {
+                        "dump"
+                    } else {
+                        &file_path
+                    },
+                    request.get_ref().raw,
+                )
+                .is_ok();
```

Comment:
I recommend that you just return "internal server error" here if you cannot start the dump.

If the string is not a valid path then return `invallid_argument`.